docs/design/01_linkage.md

```diff
@@ -0,0 +1,81 @@
+# Record linkage design
+
+## Context
+
+When a CVE comes, we have various metadata, not _ALL_ metadata are filled by authorities.
+
+What we expect to find is:
+
+- a description
+
+What we hope to find is:
+
+- a CPE
+
+## Goal
+
+Our goal is to match a CVE record with a set of derivations which we know about in all our supported channels.
+
+For this, it is sufficient to consider the latest evaluation of a given channel.
+
+We have all the derivations known in an evaluation of a given channel on one hand and the other hand a CVE record.
+
+### With CPEs
+
+Assuming we do have CPEs, we need a function to go from CPE URI to CPE WFNs.
+
+Assuming we have a CPE WFN [1] of the following form `wfn:[part=X, vendor=Y, product=Z, version=V, target_hw=TH, target_sw=TS]`.
+
+We would like to find derivations which correspond to product `Z` and vendor `Y` in nixpkgs, details on the matching are defined in [the CPE-based linkage document](./02_cpe-based_linkage.md).
+
+[1]: https://nvlpubs.nist.gov/nistpubs/Legacy/IR/nistir7695.pdf
+
+### Without CPEs
+
+To make it simple, we should do the work to attach CPEs to CVEs themselves when they are missing them.
+
+We reduce this problem to predict or infer a valid CPE albeit wrong from a text description.
```

Comment:
Not clear to me why we wouldn't jump to derivation matching directly. CPEs seem not very high fidelity information.

Reply:
Discussed with @RaitoBezarius: We're not losing anything by having an intermediate (generated or curated) CPE layer, but we gain the opportunity to verify and improve the matching quality.